packages/tupaia-web/src/features/Dashboard/DashboardMenu/SubscribeModal.tsx

```diff
@@ -3,185 +3,141 @@
+    border-bottom: 1px solid ${theme.palette.divider};
+  }`}
 `;
 
 export const SubscribeModal = () => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
   const { activeDashboard, subscribeModalOpen, toggleSubscribeModal } = useDashboard();
+  const queryClient = useQueryClient();
   const { entityCode, projectCode } = useParams();
-  const { refetch: refetchDashboards } = useDashboards(projectCode, entityCode);
   const { data: user, isLoggedIn, isLoading } = useUser();
   const mailingList = useDashboardMailingList();
   const isSubscribed = mailingList ? mailingList.isSubscribed : false;
   const defaultEmail = isLoggedIn ? user.email : '';
   const formContext = useForm({
     mode: 'onChange',
-    defaultValues: {
-      email: defaultEmail,
-    },
   });
-  const {
-    mutateAsync: subscribe,
-    error: subscribeError,
-    reset: resetSubscribe,
-  } = useSubscribeDashboard(projectCode, entityCode, activeDashboard?.code);
-  const {
-    mutateAsync: unsubscribe,
-    error: unsubscribeError,
-    reset: resetUnsubscribe,
-  } = useUnsubscribeDashboard(projectCode, entityCode, activeDashboard?.code);
+  const { mutateAsync: subscribe, error: subscribeError } = useSubscribeDashboard(
+    projectCode,
+    entityCode,
+    activeDashboard?.code,
+  );
+  const { mutateAsync: unsubscribe, error: unsubscribeError } = useUnsubscribeDashboard(
+    projectCode,
+    entityCode,
+    activeDashboard?.code,
+  );
 
   const handleSubmit = async data => {
+    setIsSubmitting(true);
     if (isSubscribed) {
       await unsubscribe(data);
     } else {
       await subscribe(data);
     }
-    refetchDashboards();
+    setIsSubmitting(false);
     toggleSubscribeModal();
+    queryClient.invalidateQueries(['dashboards']);
   };
 
-  const handleClose = () => {
-    if (isSubscribed) {
-      resetUnsubscribe();
-    } else {
-      resetSubscribe();
+  const getModalContent = () => {
+    if (isLoading) {
+      return <SpinningLoader m={5} />;
+    }
+
+    if (!activeDashboard && mailingList) {
+      return <Typography color="error">Something went wrong</Typography>;
     }
-    toggleSubscribeModal();
-  };
 
-  const isMutateError = !!subscribeError || !!unsubscribeError;
+    if (!!subscribeError || !!unsubscribeError) {
+      return (
+        <Typography color="error">
+          {isSubscribed ? unsubscribeError?.message : subscribeError?.message}
+        </Typography>
+      );
+    }
+
+    return (
+      <>
+        <ModalParagraph>
+          {isSubscribed
+            ? 'You have already joined this dashboards mailing list. If you would like to remove yourself from dashboard updates, please confirm your email below and click ‘Remove’.'
+            : 'By entering your email address you will receive an email export of this dashboard from the admin. This export will include a PDF report with the details of this dashboard. This will allow you to be notified via email when import things are happening!'}
```

Comment:
Is “import things” meant to read “important things”?

Reply:
good catch - that is coped from the design file but is clearly wrong